### PR TITLE
Fix the source of some "unused variable" warnings

### PR DESCRIPTION
**Pull request by lucafedeli88**
In this PR I have commented out some unused variables which caused several warning messages during compilation.

### CODE REVIEW

**Review by MaxThevenet — APPROVED** [review]
Looks good to me, thanks for the cleaning!

File: Source/Particles/Pusher/GetAndSetPosition.H [hunk 1]
```diff
@@ -96,7 +96,13 @@ struct SetParticlePosition
...
     AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
-    void operator() (const int i, RType x, RType y, RType z) const noexcept
+    void operator() (const int i,
+#if defined(WARPX_DIM_RZ) || defined(WARPX_DIM_3D)
+        RType x, RType y, RType z
+#else
+        RType x, RType /*y*/, RType z
+#endif
+        ) const noexcept
     {
```
Comment (ax3l) [inline Source/Particles/Pusher/GetAndSetPosition.H]: I think in this case, that would be cleaner:

```suggestion
    void operator() (const int i, RType x, RType y, RType z) const noexcept
    {
#if defined(WARPX_DIM_XZ)
        amrex::ignore_unused(y);
#endif
```

This needs to `#include <AMReX.H>` please.

**Review by ax3l — COMMENTED** [review]
Thanks a lot!
Only one small update to keep the interfaces simple :)

File: Source/Particles/Pusher/GetAndSetPosition.H (hunk 1 above)
Comment (lucafedeli88) [reply Source/Particles/Pusher/GetAndSetPosition.H]: ah ! I was not aware of this cool method!

**Review by ax3l — APPROVED** [review]
:tada:

**Pull request merged**